pkg/trait/error_handler.go

```diff
@@ -107,3 +120,17 @@ func (t *errorHandlerTrait) addGlobalErrorHandlerAsSource(e *Environment) error
 
 	return nil
 }
+
+// shouldHandleNoErrorHandler determines the runtime version and checks on noErrorHandler that is configured for this version.
+func shouldHandleNoErrorHandler(it *v1.Integration) bool {
+	if it.Status.RuntimeVersion != "" {
+		runtimeVersion, _ := strings.CutSuffix(it.Status.RuntimeVersion, "-SNAPSHOT")
+		if versionNumber, err := strconv.Atoi(strings.ReplaceAll(runtimeVersion, ".", "")); err == nil {
```

Comment:
The downstream version is something like "3.2.0.redhat-00014", I didn't check this function, but I remember having issues parsing the downstream version, so just a reminder to take this into account. We can also add this specific to the downstream fork if needed.

Reply:
We use `semver` dependency somewhere in the code. You may reuse that logic as well. In any case, I think this should be handled directly by the runtime, so that the operator doesn't know the version and we do not need to do all this hardcoding.